src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -97,64 +128,40 @@ private void initializeTranscript(Entity lectureEntity) throws IOException, Serv
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long lectureId = Long.parseLong(request.getParameter(LectureUtil.ID));
+    if (request.getParameter(PARAM_ID) == null) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, ERROR_MISSING_ID);
+      return;
+    }
+    long lectureId = Long.parseLong(request.getParameter(PARAM_ID));
     Key lectureEntityKey = KeyFactory.createKey(LectureUtil.KIND, lectureId);
     try {
       Entity lectureEntity = datastore.get(lectureEntityKey);
       Gson gson = new Gson();
       response.setContentType("application/json");
       response.getWriter().println(gson.toJson(LectureUtil.createLecture(lectureEntity)));
     } catch (EntityNotFoundException entityNotFound) {
-      response.sendError(HttpServletResponse.SC_NOT_FOUND, "Lecture not found in database.");
+      response.sendError(HttpServletResponse.SC_NOT_FOUND, ERROR_LECTURE_NOT_FOUND);
     }
   }
 
   /**
-   * Returns the Entity in database that has {@code url}, or
+   * Returns the {@code Entity} in database that has {@code videoUrl}, or
    * {@code Optional.empty()} if one doesn't exist.
    */
-  private Optional<Entity> checkUrlInDatabase(Optional<String> videoUrl) {
-    if (!videoUrl.isPresent()) {
-      return Optional.empty();
-    }
-
+  // TODO: Use a filter to avoid fetching all lectures.  See: #185.
+  private Optional<Entity> checkUrlInDatabase(String videoUrl) {
     Query query = new Query(LectureUtil.KIND);
     PreparedQuery results = datastore.prepare(query);
     Iterable<Entity> resultsIterable = results.asIterable();
 
-    String url = videoUrl.get();
     for (Entity lecture : resultsIterable) {
-      if (lecture.getProperty(LectureUtil.VIDEO_URL).equals(url)) {
+      if (lecture.getProperty(LectureUtil.VIDEO_URL).equals(videoUrl)) {
```

Comment:
Please make this its own PR. it is not related to the PR description.

Reply:
Should be gone now